targets/TARGET_STM/i2c_api.c

```diff
@@ -1023,24 +1131,34 @@ int i2c_byte_write(i2c_t *obj, int data)
         tmpreg &= ~I2C_CR2_RELOAD;
         /*  Disable Autoend */
         tmpreg &= ~I2C_CR2_AUTOEND;
-        /* Do not set any transfer size for now */
-        tmpreg |= (I2C_CR2_NBYTES & (1 << 16));
+        /*  Set transfer size to 1 */
+        tmpreg |= (I2C_CR2_NBYTES & (1 << I2C_CR2_NBYTES_Pos));
         /* Set the prepared configuration */
         handle->Instance->CR2 = tmpreg;
-    } else {
-        /* Set the prepared configuration */
-        tmpreg = handle->Instance->CR2;
 
-        /* Then send data when there's room in the TX fifo */
-        if ((tmpreg & I2C_CR2_RELOAD) != 0) {
-            while (!__HAL_I2C_GET_FLAG(handle, I2C_FLAG_TCR)) {
-                if ((timeout--) == 0) {
-                    DEBUG_PRINTF("timeout in i2c_byte_write\r\n");
-                    return 2;
-                }
+        // Wait until we get the result for the address byte.
+        // The hardware clears the I2C_CR2_START bit when the start condition has been sent.
+        timeout = BYTE_TIMEOUT;
+        while (handle->Instance->CR2 & I2C_CR2_START) {
+            if ((timeout--) == 0) {
+                return 2;
             }
         }
-        /*  Enable reload mode as we don't know how many bytes will eb sent */
+
+        // Set to read or write based on the address
+        obj_s->state = data & 0x1 ? STM_I2C_SB_READ_IN_PROGRESS : STM_I2C_SB_WRITE_IN_PROGRESS;
+
+    } else {
+
+        if(obj_s->state == STM_I2C_SB_READ_IN_PROGRESS)
+        {
```

Comment:
please keep the styling consistent, it uses `{` attached to the condition, and space after `if`, see line 1143, please fix all instances in this file

Reply:
Is there a command I can run to autoformat everything?

Reply:
Fixed, dang notepad++...